app/cdap/components/ConnectionList/index.tsx

```diff
@@ -216,13 +227,37 @@ const DatasetWrapper = () => {
 
   let headerContent;
 
+  const searchHandler = (index: number) => {
+    setDataForTabs((prev) => {
+      const tempData = [...prev];
+      tempData[index].isSearching = true;
+      return tempData;
+    });
+    refs.current[index].focus();
+  };
+
+  const handleChange = (e: any, emitter: string, index: number) => {
```

Comment:
can you put comments explaining whats going on here.

Reply:
split the code into 2 parts just to make it more readable.